### PR TITLE
Feature/Ingredients

### PR DESCRIPTION
**Pull request by rishabh-vinsol**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -18,4 +18,5 @@
   end
 
   resources :users, except: :show
+  resources :ingredients, except: :show
```
Comment (Sparsh-Goyal-2000) [inline config/routes.rb:21]: Add ingredients listing somewhere in the navbar

File: app/views/ingredients/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<%= render 'shared/notice' %>
+<div class="container mt-4">
+  <div class="border p-4">
+    <div class="d-flex justify-content-between align-items-center mb-4">
+      <h2>Ingredients</h2>
+      <%= link_to "New ingredient", new_ingredient_path, class: "btn btn-primary btn-sm" %>
+    </div>
+    <table class="table table-striped">
```
Comment (Sparsh-Goyal-2000) [inline app/views/ingredients/index.html.erb:8]: Show this table (headers) only if ingredients are present, else show message ingredients are not present.
Follow this on every listing page.

File: app/views/ingredients/_ingredient.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<tr>
+  <td><%= ingredient.id %></td>
+  <td><%= image_tag ingredient.image, height: 50%></td>
```
Comment (Sparsh-Goyal-2000) [inline app/views/ingredients/_ingredient.html.erb:3]: This is breaking is case no image is uploaded.
Fix this.

Comment (Sparsh-Goyal-2000) [reply app/views/ingredients/_ingredient.html.erb:3]: Add space after 50

File: app/models/ingredient.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class Ingredient < ApplicationRecord
+  has_one_attached :image
```
Comment (Sparsh-Goyal-2000) [inline app/models/ingredient.rb:2]: Add validations on image
size not more than 5 mb
allow only jpg/jpeg or png

File: app/views/ingredients/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    <%= form.label :price_per_portion, class: "form-label" %>
+    <%= form.number_field :price_per_portion, class: "form-control" %>
+  </div>
+  <div class="mb-3">
+    <%= form.label :non_veg, "Type ", class: "form-label" %>
+    <%= form.select :non_veg, [['Veg', false], ['Non-Veg', true]], {}, {class: "form-select"} %>
+  </div>
+  <div class="mb-3">
+    <%= form.label :image, class: "form-label" %>
+    <%= form.file_field :image, class: "form-control" %>
```
Comment (Sparsh-Goyal-2000) [inline app/views/ingredients/_form.html.erb:17]: Show image preview here while updating if already uploaded

File: app/views/ingredients/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    <%= form.label :name, class: "form-label" %>
+    <%= form.text_field :name, class: "form-control" %>
+  </div>
+  <div class="mb-3">
+    <%= form.label :price_per_portion, class: "form-label" %>
+    <%= form.number_field :price_per_portion, class: "form-control" %>
+  </div>
+  <div class="mb-3">
+    <%= form.label :non_veg, "Type ", class: "form-label" %>
+    <%= form.select :non_veg, [['Veg', false], ['Non-Veg', true]], {}, {class: "form-select"} %>
```
Comment (Sparsh-Goyal-2000) [inline app/views/ingredients/_form.html.erb:13]: formatting
`{ class: "form-select" }`

File: app/views/ingredients/_form.html.erb (hunk 5 above)
Comment (Sparsh-Goyal-2000) [inline app/views/ingredients/_form.html.erb:17]: Add `destroy_on_save` checkbox to destroy an already uploaded image while updating

File: app/views/ingredients/_form.html.erb (hunk 6 above)
Comment (Sparsh-Goyal-2000) [inline app/views/ingredients/_form.html.erb:13]: Try radio button here, it will look nice

**Review by Sparsh-Goyal-2000 — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 1 above)
Comment (rishabh-vinsol) [reply config/routes.rb:21]: Done

File: app/models/ingredient.rb (hunk 4 above)
Comment (rishabh-vinsol) [reply app/models/ingredient.rb:2]: Done

File: app/views/ingredients/_form.html.erb (hunk 5 above)
Comment (rishabh-vinsol) [reply app/views/ingredients/_form.html.erb:17]: Done

File: app/views/ingredients/_form.html.erb (hunk 6 above)
Comment (rishabh-vinsol) [reply app/views/ingredients/_form.html.erb:13]: Done

File: app/views/ingredients/index.html.erb (hunk 2 above)
Comment (rishabh-vinsol) [reply app/views/ingredients/index.html.erb:8]: Done

File: app/views/ingredients/_form.html.erb (hunk 5 above)
Comment (rishabh-vinsol) [reply app/views/ingredients/_form.html.erb:17]: Done

File: app/views/ingredients/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    <%= form.label :price_per_portion, class: "form-label" %>
+    <%= form.number_field :price_per_portion, class: "form-control" %>
+  </div>
+  <div class="mb-3">
+    <div>
+      <%= form.label :non_veg, "Type ", class: "form-label" %>
+    </div>
+    <div class="form-check form-check-inline">
+      <%= form.radio_button :non_veg, false, class: "form-check-input" %>
+      <%= form.label :non_veg_false, "Veg", class:"form-check-label" %>
```
Comment (Sparsh-Goyal-2000) [inline app/views/ingredients/_form.html.erb:17]: `Veg` should be selected by default

File: app/views/layouts/_navbar.html.erb [hunk 8]
```diff
@@ -16,23 +16,22 @@
...
-        </li>
-        <li class="nav-item">
-          <a class="nav-link" href="#">Pay</a>
-        </li>
+        <% if cookies[:auth_token] %>
+          <li class="nav-item">
+            <%= link_to 'Home', root_path, class: "nav-link active" %>
+          </li>
+          <li class="nav-item">
+            <%= link_to 'Ingredients', ingredients_path, class: "nav-link" %>
```
Comment (Sparsh-Goyal-2000) [inline app/views/layouts/_navbar.html.erb:22]: Add class `active` to the current heading.
If we are on `Ingredients` page, then `Ingredients` should be active

File: app/models/ingredient.rb [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+  validates :name, :price_per_portion, presence: true
+  validates :non_veg, inclusion: { in: [true, false] }
+  validate :image_file_size
```
Comment (Sparsh-Goyal-2000) [inline app/models/ingredient.rb:6]: We can use active storage validations here.

File: app/controllers/ingredients_controller.rb [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @ingredient.update(ingredient_params)
+      if params[:ingredient][:destroy_on_save] == '1'
+        @ingredient.image.purge
+      end
```
Comment (Sparsh-Goyal-2000) [inline app/controllers/ingredients_controller.rb:26]: We can use single line if here.
Aren't you following rubocop rules?
First, take out a branch from main and fix all rubocops.

**Review by Sparsh-Goyal-2000 — CHANGES_REQUESTED** [review]

**Review by Sparsh-Goyal-2000 — CHANGES_REQUESTED** [review]

**Review by Sparsh-Goyal-2000 — APPROVED** [review]

**Pull request merged**